### PR TITLE
unicode: update for modules, Unicode 15.0.0

### PR DESCRIPTION
**Pull request by rsc**
- $GOPATH search doesn't work with modules. Instead, embed the UnicodeData.txt from Unicode.
- Derive unicode.txt content as needed instead of having separate copy.
- Update UnicodeData.txt to Unicode 15.0.0, adding a go generate line.

### CODE REVIEW

File: unicode.go [hunk 1]
```diff
@@ -14,23 +14,15 @@ usage: unicode [-c] [-d] [-n] [-t]
...
-	ftp://ftp.unicode.org/Public/UNIDATA/UnicodeData.txt
-to
-	$GOPATH/src/robpike.io/cmd/unicode
 */
 package main // import "robpike.io/cmd/unicode"
 
 import (
-	"bufio"
 	"bytes"
+	_ "embed"
```
Comment (robpike) [inline unicode.go:22]: this being a different class of thing, i'd expect it to be its own paragraph inside the import list, like an external module. but gofmt doesn't do that. humph.

File: unicode.go [hunk 2]
```diff
@@ -44,75 +36,32 @@ var (
 	doUnic = flag.Bool("u", false, "describe the characters from the Unicode database, in Unicode form")
 	doUNIC = flag.Bool("U", false, "describe the characters from the Unicode database, in glorious detail")
 	doGrep = flag.Bool("g", false, "grep for argument string in data")
-
-	doPrintUnicodeTxt = flag.Bool("db", false, "convert UnicodeData.txt to unicode.txt format on stdout")
 )
 
 var printRange = false
 
+//go:generate sh -c "curl http://ftp.unicode.org/Public/UNIDATA/UnicodeData.txt >UnicodeData.txt"
```
Comment (robpike) [inline unicode.go:43]: this won't work on windows or plan 9 or another system without curl.

instead put this in the README, and the user can get it how they want.

**Review by robpike — CHANGES_REQUESTED** [review]

**Comment by robpike** [comment]
P.S. thanks, i was planning to do this myself.

File: unicode.go (hunk 2 above)
Comment (rsc) [reply unicode.go:43]: The user won't be running this - only you will, when you want to issue an update.

Comment (robpike) [reply unicode.go:43]: ah of course.

**Pull request merged**